actionview/lib/action_view/template/render_in_object.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module ActionView # :nodoc:
+  class Template # :nodoc:
+    class RenderInObject # :nodoc:
+      attr_accessor :type
+
+      def initialize(component)
+        @component = component
+      end
+
+      def identifier
+        @component.class.name
+      end
+
+      alias_method :inspect, :identifier
+
+      def render(context, *args)
+        @component.render_in(context)
+      end
+
+      def format
+        @component.format
+      end
+
+      def formats; Array(format); end
+      deprecate :formats
```

Comment:
I see that this file is very similar to html and text. Maybe this needs a better name and a parent class to inherit from so we can reduce some of the duplication.

This looked to me like a brand new class that was deprecating a brand new method. I think there is some room for an improvement to the template object.

Reply:
@eileencodes makes sense. Perhaps we should fix the other ActionView template objects, then? They have the same :nodoc: pattern I used in this PR. I'd be happy to write the PR ❤️